avocado/plugins/spawners/remote.py

```diff
@@ -0,0 +1,210 @@
+import asyncio
+import contextlib
+import json
+import logging
+import os
+import shlex
+
+from aexpect import remote
```

Comment:
Avocado doesn't have a dependency on aexpect.  So we have two options (in order of preference):

1. Make it an optional plugin
2. Protect against import errors (and when/if https://github.com/avocado-framework/avocado/pull/5649 is merged, signal that in `is_operational()`)

Reply:
Indeed, I can do something similar to the LXC_AVAILABLE where I also suggested it gets into the `is_operational` method [here](https://github.com/avocado-framework/avocado/pull/5649/files#r1271447476). I think then I will wait for the above pull request to get merged and will update this pull request accordingly right after.

Reply:
So, the `is_operational` method will go a long way and will probably solve this.  But, depending on another external library in a "built in" plugin would:
* require either an "import protection" (try/import/except)
* require development mode of the core avocado to have aexpect

IMO, the best solution here is to have this plugin (and any other with extra Python package deps) as an optional_plugin.  I tried to come up with an example, and ended up with this working version: https://github.com/clebergnu/avocado/commits/pevogam_remote_spawner_review

Please consider that approach, and feel free to cherry-pick those commits.

Reply:
Alright, I am not sure about your opinion about having an optional LXC plugin then and the comments above about refactoring code from an optional and core plugin (or perhaps two optional plugins) seems harder if at least one of them is optional. Where would you recommend we refactor functionality as in your comment https://github.com/avocado-framework/avocado/pull/5621#discussion_r1267276198 and https://github.com/avocado-framework/avocado/pull/5621#discussion_r1267278540? If you prefer of course we might as well not refactor anything from the plugins yet until both of them are fully stabilized over the longer term and simply end this with your effort to migrate the remote spawner to an optional plugin.